src/segment-loader.js

```diff
@@ -1176,31 +1169,33 @@ export default class SegmentLoader extends videojs.EventTarget {
       return null;
     }
 
+    let segmentMediaIndex = null;
```

Comment:
I think it may be helpful to distinguish between `mediaIndex` and `segmentMediaIndex`. Maybe `mediaIndex` can be `currentMediaIndex` and `segmentMediaIndex` can be `mediaIndex`, or `segmentMediaIndex` can be something like `nextMediaIndex`.